scripts/create_gitlab_snapshot.py

```diff
@@ -0,0 +1,107 @@
+import datetime
+import logging
+from time import (
+    sleep,
+)
+from typing import (
+    Mapping,
+)
+
+import botocore.exceptions
+from more_itertools import (
+    one,
+)
+
+from azul import (
+    JSON,
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+
+def main():
+    require(config.terraform_component == 'gitlab',
+            "Select the 'gitlab' component ('dev.gitlab' or 'prod.gitlab', for example).")
+    instance = gitlab_instance_info()
+    shutdown_instance(instance)
+    create_snapshot(instance)
+
+
+def gitlab_instance_info() -> Mapping[str, JSON]:
+    filters_ = [
+        {
+            'Name': 'key-name',
+            'Values': ['azul-gitlab']
+        }
+    ]
+    return one([
+        one(reservation['Instances'])
+        for reservation in aws.ec2.describe_instances(Filters=filters_)['Reservations']
+    ])
+
+
+def shutdown_instance(instance: Mapping[str, JSON]):
+    instance_ids = [instance['InstanceId']]
+    log.info('Preparing to stop GitLab instance for %s, waiting 10 seconds before proceeding',
+             config.deployment_stage)
+    sleep(10)
+    log.info('Stopping instance %s', instance_ids)
+    aws.ec2.stop_instances(InstanceIds=instance_ids)
+    try:
+        log.info('Waiting up to 10 minutes for the instance to stop')
+        waiter = aws.ec2.get_waiter('instance_stopped')
+        waiter.wait(InstanceIds=instance_ids)
+    except botocore.exceptions.WaiterError as e:
+        log.warning(e)
```

Comment:
Why would we want to ignore this exception? Taking a snpashot on a running instance will very likely produce a snapshot with file system errors.